### PR TITLE
Urlencode & append extension to channel logo set from channel name for remote path logo source

### PR DESCRIPTION
**Pull request by allixx**
https://github.com/kodi-pvr/pvr.iptvsimple/issues/210

### CODE REVIEW

File: src/client.h [hunk 1]
```diff
@@ -50,7 +53,9 @@ extern bool        g_bTSOverride;
 extern bool        g_bCacheM3U;
 extern bool        g_bCacheEPG;
 extern int         g_iEPGLogos;
+extern int         g_logoPathType;
 
 extern std::string PathCombine(const std::string &strPath, const std::string &strFileName);
 extern std::string GetClientFilePath(const std::string &strFileName);
 extern std::string GetUserFilePath(const std::string &strFileName);
+extern const std::string UrlEncode(const std::string &value);
```
Comment (afedchin) [inline src/client.h]: you can declare `UrlEncode` method where it's used i.e. in `PVRIptvData.cpp` file. no need to export it

**Review by afedchin — CHANGES_REQUESTED** [review]

**Comment by allixx** [comment]
I'm not sure what jenkins problem actually is.

**Comment by afedchin** [comment]
jenkins build this please

File: src/PVRIptvData.cpp [hunk 2]
```diff
@@ -77,13 +77,36 @@ inline bool GetAttributeValue(const xml_node<Ch> * pNode, const char* strAttribu
   return true;
 }
 
+// http://stackoverflow.com/a/17708801
+static const std::string UrlEncode(const std::string &value) {
```
Comment (afedchin) [inline src/PVRIptvData.cpp]: please change your code style according to the guidelines https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#31-braces

**Review by afedchin — CHANGES_REQUESTED** [review]

File: src/PVRIptvData.cpp [hunk 3]
```diff
@@ -77,13 +77,39 @@ inline bool GetAttributeValue(const xml_node<Ch> * pNode, const char* strAttribu
...
 }
 
+// http://stackoverflow.com/a/17708801
+static const std::string UrlEncode(const std::string &value)
+{
+  std::ostringstream escaped;
+  escaped.fill('0');
+  escaped << std::hex;
+
+  for (std::string::const_iterator i = value.begin(), n = value.end(); i != n; ++i)
```
Comment (phunkyfish) [inline src/PVRIptvData.cpp:87]: Why not: ```for (auto c : value)``` and remove ```std::string::value_type c = (*i);``` completely?

File: src/PVRIptvData.cpp [hunk 4]
```diff
@@ -426,6 +454,12 @@ bool PVRIptvData::LoadPlayList(void)
         tmpChannel.iTvgShift  = (int)(fTvgShift * 3600.0);
         tmpChannel.bRadio     = bRadio;
 
+        // urlencode channel logo when set from channel name and source is Remote Path
+        // append extension as channel name wouldn't have it
+        if (m_logoPathType == 1 && logoFromChannelName)
+          tmpChannel.strTvgLogo = \
```
Comment (phunkyfish) [inline src/PVRIptvData.cpp:460]: Just put this on one line

File: src/PVRIptvData.h [hunk 5]
```diff
@@ -23,7 +23,9 @@
  *
  */
 
+#include <iomanip>
 #include <map>
+#include <sstream>
```
Comment (phunkyfish) [inline src/PVRIptvData.h:28]: I think can be removed

File: src/PVRIptvData.h [hunk 6]
```diff
@@ -23,7 +23,9 @@
  *
  */
 
+#include <iomanip>
```
Comment (phunkyfish) [inline src/PVRIptvData.h:26]: Should be in cpp file and not the header.

**Comment by phunkyfish** [comment]
@allixx are you still interested in finishing this PR?

**Comment by allixx** [comment]
Sorry. 
Looks like thanks to your impressive refactor work all required bits are already there, and all that's left is to add the urlencode function (could it somehow be pulled from kodi api by the way?) and url formatting conditional statement in playlist loader's channel parser, right?

**Comment by phunkyfish** [comment]
So this would need to be done in the Leia branch as that was not refactored.

And then again in the Matrix branch. Or you could just do it in Matrix. Up to you.

**Comment by allixx** [comment]
I seem to remember kso saying they didn't want any new features in Leia, so this probably goes to Matrix.
Can you do all the required changes by yourself, if you are up to it of course, it's going to be much quicker and I don't want to waste any more of your time reviewing my mistakes.

**Comment by phunkyfish** [comment]
Sure, no problem. If you can test it in Matrix I will add to next PR?

**Comment by allixx** [comment]
Yes, I can help with testing I guess. Cannot promise instant reaction time though...

**Comment by phunkyfish** [comment]
No rush.

**Pull request closed without merging**